modules/auxiliary/gather/zookeeper_info_disclosure.rb

```diff
@@ -0,0 +1,80 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::Scanner
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'Apache ZooKeeper Information Disclosure',
+      'Description' => %q{
+        Apache Zookeeper server service runs on TCP 2181 and by default, it is accessible without any authentication. This module targets Apache ZooKeeper service instances to extract information about the system environment, and application configuration.
+      },
+      'References' =>
+        [
+          ['URL', 'https://zookeeper.apache.org/doc/current/zookeeperAdmin.html']
+        ],
+      'Author' =>
+        [
+          'Karn Ganeshen <KarnGaneshen[at]gmail.com>'
+        ],
+      'DisclosureDate' => 'Oct 14, 2020',
+      'License' => MSF_LICENSE,
+      'DefaultOptions' => { 'VERBOSE' => true })
+      )
+
+    register_options(
+      [
+        Opt::RPORT(2181),
+        OptInt.new('TIMEOUT', [true, 'Timeout for the probe', 30])
+      ], self.class
+    )
+
+    deregister_options('USERNAME', 'PASSWORD')
+  end
+
+  def run_host(ip)
+    to = (datastore['TIMEOUT'].zero?) ? 30 : datastore['TIMEOUT']
```

Comment:
I agree that we shouldn't have an infinite timeout for networking operations, but I do think we should inform the user that the timeout value has been overridden if they set it zero, even though it looks like this code was copied from elsewhere.

Reply:
I've now added this after line 41: `vprint_status("Using a timeout of #{to}...")`. If a user sets the timeout to 0, this is the run log:
```
[*] 1.3.3.7:2180    - Using a timeout of 30...
[-] 1.3.3.7:2180    - 118.190.161.4:2180 - Connection Failed...
[*] 1.3.3.7:2180    - Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
```

Reply:
I've now added this after line 41: `vprint_status("Using a timeout of #{to}...")`. If a user sets the timeout to 0, this is the run log:
```
[*] 1.3.3.7:2180    - Using a timeout of 30...
[-] 1.3.3.7:2180    - 1.3.3.7:2180 - Connection Failed...
[*] 1.3.3.7:2180    - Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
```